### DISCUSSION TITLE
apollo会覆盖application.properties吗

### DISCUSSION THREAD

**Discussion by jiangxiaoqiang** [apolloconfig/apollo#4029] [Q&A] [answered]
在使用的过程中感觉apollo配置中心的配置会覆盖application.properties的配置，如果在本地应用application.properties有此配置，而apollo配置中心没有，那么最终运行是用apollo配置中心的配置么？ 不会取配置中心与本地application.properties的并集？不是很确定。

**Comment by nobodyiam** [comment, marked as answer]
可以看下 [3.1 和Spring集成的原理](https://www.apolloconfig.com/#/zh/design/apollo-design?id=_31-%e5%92%8cspring%e9%9b%86%e6%88%90%e7%9a%84%e5%8e%9f%e7%90%86)

**Comment by Anilople** [comment]
根据源码 https://github.com/apolloconfig/apollo/blob/a89d081bac735af3392a65cb6a6b762f182b608c/apollo-client/src/main/java/com/ctrip/framework/apollo/spring/boot/ApolloApplicationContextInitializer.java#L144
apollo的namespace `application` 会覆盖 本地的 `application.properties`

---

According source https://github.com/apolloconfig/apollo/blob/a89d081bac735af3392a65cb6a6b762f182b608c/apollo-client/src/main/java/com/ctrip/framework/apollo/spring/boot/ApolloApplicationContextInitializer.java#L144
apollo's namespace `application` will cover local `application.properties`
